consensus/src/marshal/ingress/mailbox.rs

```diff
@@ -84,6 +85,22 @@ pub(crate) enum Message<S: Scheme, B: Block> {
         /// A channel to send the retrieved finalization.
         response: oneshot::Sender<Option<Finalization<S, B::Commitment>>>,
     },
+    /// A request to ensure a finalization is available for a given height.
+    ///
+    /// Unlike [Self::GetFinalization] which only checks local storage, this triggers
+    /// a network fetch if the finalization is not available locally. This is
+    /// fire-and-forget, the finalization will be stored in marshal and delivered
+    /// via the normal finalization flow when available.
+    ///
+    /// Targets are required because this is typically called when a peer claims to
+    /// be ahead. By targeting only that peer, we can block them if they don't
+    /// actually have the finalization.
+    EnsureFinalization {
```

Comment:
I think the name of this is misleading...I think we could call it `UntrustedFinalization` or `Hint` or something?